src/types.ts

```diff
@@ -13,4 +13,42 @@
 * under the License.
 */
 
-export type HttpMethod = "GET" | "POST";
\ No newline at end of file
+// HTTP requests & responses related 
+export type HttpMethod = "GET" | "POST";
+export interface ListCount {count: number}
+export interface ResponseList<T>  {
+    nextPaginationToken?: string
+    status: Response['status'],
+    users: T[]
+}
+
+
+// Users Models
+export enum UserRecipeType { 
+  emailpassword= 'emailpassword', 
+  thirdparty = "thirdparty", 
+  passwordless = "passwordless" }
+
+export type User = { recipeId: UserRecipeType.emailpassword, user: UserEmailPassword } | 
+  { recipeId: UserRecipeType.passwordless, user: UserPasswordLess } |
+  { recipeId: UserRecipeType.thirdparty, user: UserThirdParty }
+
+export interface UserProperties {
+  id: string
+  email?: string
+  phoneNumber?: string
+  timeJoined?: number
+  firstName?: string
+  lastName?: string
+}
+
+export type UserEmailPassword = UserProperties
+
+export type UserPasswordLess = UserProperties
+
+export type UserThirdParty = UserProperties & {
+  thirdParty: {
+    id: 'google' | 'github' | string
```

Comment:
Missing apple and facebook (https://github.com/supertokens/supertokens-node/tree/master/lib/ts/recipe/thirdparty/providers)